### PR TITLE
Partially update user guide

### PR DESCRIPTION
**Pull request by ianyong**
Changes:
- Add section 2.1 on the layout of the UI.
  - Add images of all 4 tabs.
- Add preface for section 4.2 of tabs.
- Add `delete` command for `Expense` and `Income`.
- Add newlines after headings and subheadings.
- Fix links to section 2.1.
  - See https://github.com/AY2021S1-CS2103T-W16-3/tp/pull/178#discussion_r511513961.
- Remove bullet points for table of contents.

### CODE REVIEW

File: docs/_sass/minima/_layout.scss [hunk 1]
```diff
@@ -261,3 +261,12 @@
     width: calc(50% - (#{$spacing-unit} / 2));
   }
 }
+
+
+
```
Comment (ianyong) [inline docs/_sass/minima/_layout.scss:266]: Believe it or not, the 3 line breaks here is for consistency with the rest of the file.

**Review by yongping827 — APPROVED** [review]
LGTM

**Review by zhaojj2209 — APPROVED** [review]
LGTM!

**Comment by ianyong** [comment]
Related to #131.

**Pull request merged**